File: python/40d_1669B_triple.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        n = int(input())
        arr = input().split()
        hmap = {}
        num = -1
        for i in arr:
            if i not in hmap:
                hmap[i] = 1
            else:
                hmap[i] += 1
            if hmap[i] >= 3:
                num = i
        ans.append(num)
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/40d_1669B_triple.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, temp, num;
    std::vector<int> ans;
    std::cin>>t;
    for (int i = 0; i < t; i++){
        std::cin>>n;
        std::unordered_map<int, int> hmap;
        num = -1;
        for(int i = 0; i < n; i++){
            std::cin>>temp;
            hmap[temp]++;
            if(hmap[temp] >= 3){
                num = temp;
            }
        }
        ans.push_back(num);
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}